enhancements/baremetal/an-slo-for-baremetal.md

```diff
@@ -0,0 +1,308 @@
+2. Implement cluster-level configuration under `config.openshift.io`
+   (as per openshift/api#124)
+3. Expose a `/metrics` endpoint for Prometheus to be configured to
+   scrape (via a `ServiceMontitor`) and define any relevant Prometheus
+   alert rules based on those metrics.
+
+### "Not In Use" SLO Behaviors
+
+Unlike most other SLOs, the BMO is not applicable to all cluster
+configurations. On clusters running on an infrastructure platform
+other than `baremetal` it should adhere to the emerging expected
+behaviors for "not in use" SLOs, including:
+
+1. Setting its `ClusterOperator` condition to `Disabled=true`,
+  `Available=true`, `Progressing=False` with appropriate messages.
+2. User interfaces should convey this disabled state differently than
+   a failure mode (e.g. by graying it out).
+
+Currently, insights-operator is the only other example of an SLO
+following this pattern of using a `Disabled` status. Other somewhat
+similar cases following different patterns include:
+
+- Image registry and samples have a `Removed` management state where
+  `Degraded=False`, `Progressing=False`, `Available=True` with
+  `Reason=currentlyUnmanaged`.
+- The cluster credentials operator has a `disabled` config map setting
+  that can be used to disable the operator and it then sets
+  ClusterOperator status conditions to `Degraded=False`,
+  `Progressing=False`, `Available=True` with
+  `Reason=OperatorDisabledByAdmin` for all three conditions.
+
+### BMO Details
+
+The BMO will:
+
+- Be a new `openshift/cluster-baremetal-operator` project.
+- Publish an image called `cluster-baremetal-operator`.
+- Use CVO run-level 30, so its manifests will be applied in parallel
+  with the MAO.
+- Add a new `baremetal` `ClusterOperator` with an additional
+  `Disabled` status for non-baremetal platforms.
+- Use a new namespace called `openshift-baremetal`.
+- Install the `metal3.io` `Provisioning` (cluster-scoped) and
+  `BareMetalHost` (namespaced) CRDs.
+- Run under a new `openshift-baremetal/cluster-baremetal-operator`
+  `ServiceAccount`.
+- Be launched by a `openshift-baremetal/cluster-baremetal-operator`
+  `Deployment`, copying much of the MAO pod spec in terms of
+  `system-node-critical` priority class, running on masters, security
+  context, resource requests, etc.
+- Implement a controller reconciling the singleton
+  `provisioning-configuration` cluster-scoped `Provisioning` resource
+- Do nothing except set `Disabled=true`, `Available=true`, and
+  `Progressing=False` when the `Infrastructure` resource a platform
+  type other than `BareMetal`.
+- Based on the values in the `Provisioning` resource, create a
+  `metal3` `Deployment` and associated `metal3-mariadb-password` under
+  the `openshift-baremetal` namespace. This is the same as the MAO
+  currently creates under the `openshift-machine-api` namespace.
+
+### Test Plan
+
+### Graduation Criteria
+
+### Upgrade / Downgrade Strategy
```

Comment:
I guess here we some how have to deal with removing the logic from MAO towards BMO

Reply:
Yeah that's a good point, we'll have to either kill and restart any existing pod, or perhaps enable adoption of an existing metal3 pod and associated resources?

Reply:
We should be able to adopt the existing `metal3` deployment. See the new upgrade/downgrade strategy section.